include/circt/Firtool/Firtool.h

```diff
@@ -218,6 +219,38 @@ struct FirtoolOptions {
           "address conflict behavivor of combinational memories"),
       llvm::cl::init(false), llvm::cl::cat(category)};
 
+  //===----------------------------------------------------------------------===
+  // External Clock Gate Options
+  //===----------------------------------------------------------------------===
+
+  seq::ExternalizeClockGateOptions clockGateOpts;
+
+  llvm::cl::opt<std::string, true> ckgModuleName{
+      "ckg-name", llvm::cl::desc("Clock gate module name"),
+      llvm::cl::location(clockGateOpts.moduleName),
+      llvm::cl::init("EICG_wrapper"), llvm::cl::cat(category)};
+
+  llvm::cl::opt<std::string, true> ckgInputName{
+      "ckg-input", llvm::cl::desc("Clock gate input port name"),
+      llvm::cl::location(clockGateOpts.inputName), llvm::cl::init("in"),
+      llvm::cl::cat(category)};
+
+  llvm::cl::opt<std::string, true> ckgOutputName{
+      "ckg-output", llvm::cl::desc("Clock gate output port name"),
+      llvm::cl::location(clockGateOpts.outputName), llvm::cl::init("out"),
+      llvm::cl::cat(category)};
+
+  llvm::cl::opt<std::string, true> ckgEnableName{
+      "ckg-enable", llvm::cl::desc("Clock gate enable port name"),
+      llvm::cl::location(clockGateOpts.enableName), llvm::cl::init("en"),
+      llvm::cl::cat(category)};
+
+  llvm::cl::opt<std::string, true> ckgTestEnableName{
+      "ckg-test-enable",
+      llvm::cl::desc("Clock gate test enable port name (optional)"),
```

Comment:
I'm wondering whether there is a better way to handle this. I think practically it looks fine as is, but maybe? in the feature we might want to mix different flavors of clock gates. Another direction would be to introduce dedicated attributes into clock gate ops regarding mapped modules but even with that direction it'll be necessary to add similar options to LowerToHW. So probably the current implementation would be simpler. I guess this is a general problem about intrinsic lowering (how to map intrinsic to external modules, legalize to SV/Comb constructs) so perhaps we might want to revisit in the future.

Reply:
Yeah that's a great point; we'll want to revisit this sometime down the road to see if we need more localized lowering specifications. For example, being able to specify a clock gate for some subhierarchy of the design.

My rationale for the current approach is that we currently have no options and no lowering, and this introduces at least a global option and some lowering. Hopefully we can expand this in the future!

Reply:
Is there/does there need to be support for the clock divider DFT signal thing too? cc #5219 . Haven't paged this in yet, apologies if that doesn't make sense :+1: .

Reply:
Yeah good point, that's definitely something that needs to go into a follow-up PR. Right now this just handles emission of the `seq.clock_gate` op, which lowers from the `firrtl.int.clock_gate` intrinsic. The FIRRTL pipeline doesn't support that intrinsic yet in its passes (ExtractTestCode, WireDFT, etc.) -- on my todo list 😏 I totally expect that there will be some extensions to the intrinsic or `seq.clock_gate` op definitions.